### ISSUE TITLE
P0 REGRESSION + heterogeneity: champion h=828 → BPB=12.50 @ step=26K, but WAVE2-ULTRA same config → BPB=0.0002 @ step=20K

### ISSUE THREAD

**Issue by gHashTag** [gHashTag/trios-trainer-igla#51]
# 🚨 P0 REGRESSION — champion repro h=828 lr=0.0004 ≫ architectural floor 2.19

## Evidence (verbatim, this session, fresh from `bpb_samples`)

The R5 follow-up probe `IGLA-CHAMPION-REPRO-FP32-h828-step27000-rng1597` (classic ALPHA champion config: h=828, ctx=12, lr=0.0004) is now producing **real-looking** monotonic BPB curve — but **5×–6× above the architectural floor**.

```
step=22000   bpb=12.8981   ts=2026-04-30T00:48Z
step=23000   bpb=12.9573
step=24000   bpb=12.7076
step=25000   bpb=12.6052
step=26000   bpb=12.4965   ts=2026-04-30T00:52Z
```

Expected for this config (per [trios#143 champion](https://github.com/gHashTag/trios/issues/143), [trios@2446855](https://github.com/gHashTag/trios/commit/2446855)): `BPB → 2.19 ± 0.01` by step=27000.

Observed: `BPB ≈ 12.50` at step=26000, decreasing slowly. **Gap to floor: ~10.3 BPB points.**

## Two updates from anomaly issue [trios-railway#97](https://github.com/gHashTag/trios-railway/issues/97)

- **Eval bug NOT total** — long runs (step≥20000) report realistic monotonic BPB. Loss curve is real.
- **step=1000 short runs ARE overfit artifact** on tiny corpus — 105 of 114 suspect rows confirmed via `bpb_samples WHERE step≥1000 AND bpb<0.02` (last 30 min).

## Hypothesis — root cause is regression in commit `69b2d72`

`69b2d72` was the CLI alignment fix in `bin/seed-agent/src/trainer.rs` that aligned `--seed/--steps/--hidden/--lr` with `trios-train`. While the args are now correct, **the training trajectory is ~6× off floor.** Suspect:

1. `trios-train` lr schedule does NOT match the historical champion's. Champion may have used warmup or cosine decay that current `--lr 0.0004` flag bypasses.
2. Optimizer state (Adam β₁/β₂/ε) may have shifted in the trainer SOT migration ([trios-trainer-igla#4](https://github.com/gHashTag/trios-trainer-igla/issues/4)).
3. Tokenizer / corpus loader could be loading a different corpus than the historical champion run.
4. `hybrid_attn` 2-layer config wiring may have drifted ([trios-trainer-igla#5](https://github.com/gHashTag/trios-trainer-igla/issues/5) JEPA migration).

## Required (P0 — blocks Gate-2 deadline at T-23h)

1. Maintainer of `trios-train` to compare current training loop end-to-end vs the known champion run. Specifically: lr_schedule, optimizer params, corpus checksum, attn wiring.
2. Add CI smoke `tests/champion_smoke.rs` that asserts `final_bpb < 3.0` at step=10000 on h=828 config.
3. If this is irreparable in 16 h: pin `trios-train` to a pre-69b2d72 SHA in `seed-agent` and accept that BPB-eval may report wrong values at step<2000 but yield realistic curves at step≥4000.

## Triplet (R7)

```
BPB=12.4965 @ step=26000 seed=1597 sha=69b2d72 jsonl_row=N/A gate_status=REGRESSION-P0
```

## Refs

- Anomaly thread: [trios-railway#97](https://github.com/gHashTag/trios-railway/issues/97)
- Trainer eval semantics question: [trios-trainer-igla#50 comment](https://github.com/gHashTag/trios-trainer-igla/issues/50#issuecomment-4348581118)
- Champion baseline: [trios@2446855](https://github.com/gHashTag/trios/commit/2446855) BPB=2.2393
- Skill: `tri-gardener-runbook` §"Architectural BPB floor"

phi^2 + phi^-2 = 3 · TRINITY · NEVER STOP

**Comment by gHashTag** [comment]
## R5 DIAGNOSIS — eval is NOT broken; we were confusing under-trained CE with broken eval

Fresh evidence after WAVE4 mirror runs (60 reproductions of champion config):

| seed | step | bpb | repro count |
|---|---|---|---|
| 1597 | 27000 | **12.4148** | 6/6 (acc0-acc5, identical to 6 decimals) |
| 2584 | 27000 | **16.8657** | 2/2 (acc1+acc4, identical) |
| 1597 | 15000 | 0.0065 | 1 (acc5) |

**Determinism confirmed.** Same seed × same step × same config → byte-identical BPB across 6 different Railway accounts. Eval is NOT non-deterministic.

**Root insight (R5 honest):** historical champion BPB=2.1919 was measured at **step=81000** (per [trios#143](https://github.com/gHashTag/trios/issues/143)), NOT at step=27000. We've been re-running for 27K and reading "still high" as broken. It's not broken — it's **under-trained**.

| Step | Expected BPB (from champion lineage) |
|---|---|
| 1000 | ~5–10 (suspect rows reading <0.02 are short-run artifact, possibly train-loss-on-mini-batch) |
| 15000 | ~3–5 (unverified) |
| 27000 | ~3.5 (we observe 12.4 — still significantly above) |
| 50000 | ~2.5 |
| **81000** | **2.19 (champion)** |

The 12.4 → 2.19 trajectory means we need **3× more training compute** than what the WAVE2-ULTRA step=20000 runs gave us. Hackathon Gate-2 deadline = 23:59 UTC = T-21h. Even at full 6-fleet parallelism, hitting 81K steps × 3 sanctioned seeds within 21h is **borderline feasible** if we abandon all step<10000 experiments and concentrate compute.

## Updated triplet (R7)

```
BPB=12.4148 @ step=27000 seed=1597 sha=69b2d72 jsonl_row=N/A gate_status=UNDER-TRAINED-NOT-BROKEN
```

## Required (downgrade from REGRESSION-P0 to TRAINING-COMPUTE-BOUND)

1. **Cancel all WAVE2/WAVE3 step<10000 experiments** in `experiment_queue` — they cannot reach Gate-2.
2. **Spawn 18 long-runs** (3 sanctioned seeds × 6 accounts × step=81000) at priority=99.
3. **Accept R5-honest verdict:** Gate-2 likely missed at deadline UNLESS we also pin down why step=1000 reads bpb<0.02 (still real bug, but lower priority — possibly trainer eval at step=1000 just stochastic on micro-batch).

Sergeant→COLONEL contract still BLOCKED until 1 seed BPB<1.85 step≥4000. With current trajectory (BPB=12.4 at step=27K, slope ~ 0.04 BPB/1000-steps), we'd need **step ≥ 270K** to cross 1.85 — **not achievable** by deadline. R5: Gate-2 will MISS HONESTLY today.

## Refs

- [trios-trainer-igla#50](https://github.com/gHashTag/trios-trainer-igla/issues/50)
- [trios-trainer-igla#51](https://github.com/gHashTag/trios-trainer-igla/issues/51) ← downgrade
- [trios-railway#97](https://github.com/gHashTag/trios-railway/issues/97)
- Champion baseline: [trios@2446855](https://github.com/gHashTag/trios/commit/2446855)

phi^2 + phi^-2 = 3 · TRINITY · NEVER STOP
